splunk/resource_splunk_admin_saml_groups.go

```diff
@@ -100,7 +122,7 @@ func adminSAMLGroupsUpdate(d *schema.ResourceData, meta interface{}) error {
 
 func adminSAMLGroupsDelete(d *schema.ResourceData, meta interface{}) error {
 	provider := meta.(*SplunkProvider)
-	resp, err := (*provider.Client).DeleteAdminSAMLGroups(d.Id())
+	resp, err := (*provider.Client).DeleteAdminSAMLGroups(d.Get("name").(string))
```

Comment:
Because the legacy functions expected the Id to be the name, avoid potential issues caused by inconsistency between the two providers by always using the `name` when updating/deleting SAML groups.